tests/core/test_config.py

```diff
@@ -477,12 +482,43 @@ def test_variables():
         "UPPERCASE_VAR": 2,
     }
     config = Config(
-        variables=variables, gateways={"local": GatewayConfig(variables=gateway_variables)}
+        variables=variables,
+        gateways={"local": GatewayConfig(variables=gateway_variables)},
     )
     assert config.variables == variables
     assert config.get_gateway("local").variables == {"uppercase_var": 2}
 
     with pytest.raises(
-        ConfigError, match="Unsupported variable value type: <class 'sqlglot.expressions.Column'>"
+        ConfigError,
+        match="Unsupported variable value type: <class 'sqlglot.expressions.Column'>",
     ):
         Config(variables={"invalid_var": exp.column("sqlglot_expr")})
+
+
+def test_load_duckdb_attach_config(tmp_path_factory):
+    config_path = tmp_path_factory.mktemp("yaml_config") / "config_duckdb_attach.yaml"
```

Comment:
Since this is function scoped, you can do `tmp_path` which will give you a unique folder path for this function. Then this line would change to `tmp_path / "config_duckdb_attach.yaml"`